Utility Mods/Stable/SCDefenseBlocks/Data/Scripts/OneFuckingFolderDeeper/FieldGenerator/HeartNetworking/PacketQueueManager.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ProtoBuf;
+using Starcore.FieldGenerator.Networking.Custom;
+
+namespace Starcore.FieldGenerator.Networking
+{
+    public class PacketQueueManager
+    {
+        public static PacketQueueManager I;
+
+        private Dictionary<long, LinkedList<FloatSyncPacket>> floatPacketQueues = new Dictionary<long, LinkedList<FloatSyncPacket>>();
+        private Dictionary<long, LinkedList<IntSyncPacket>> intPacketQueues = new Dictionary<long, LinkedList<IntSyncPacket>>();
+        private Dictionary<long, LinkedList<BoolSyncPacket>> boolPacketQueues = new Dictionary<long, LinkedList<BoolSyncPacket>>();
+
+        private List<long> entityIds = new List<long>();
+        private int lastEntityProcessedIndex = -1;
+
+        public void Init()
+        {
+            I = this;
+        }
+
+        public void Close()
+        {
+            I = null;
+        }
+
+        // Seperate Enqueues needed because reasons (TODO: Be Smarter)

```

Comment:
if you wanted to be a turbonerd could always make use of inheritance, but if the current method works in testing then send it